libtool.m4: Change libtool to handle sysroots by default

Rather than using no sysroot by default, always query gcc to obtain the sysroot.

Signed-off-by: Richard Purdie <richard.purdie@linuxfoundation.org>

Upstream-Status: Inappropriate [Upstream are unlikely to accept this change of default]

---
 m4/libtool.m4 | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/m4/libtool.m4 b/m4/libtool.m4
index 2f31d24..bd90775 100644
--- a/m4/libtool.m4
+++ b/m4/libtool.m4
@@ -1254,18 +1254,18 @@ dnl lt_sysroot will always be passed unquoted.  We quote it here
 dnl in case the user passed a directory name.
 lt_sysroot=
 case $with_libtool_sysroot in #(
- yes)
+ no)
    if test yes = "$GCC"; then
      # Trim trailing / since we'll always append absolute paths and we want
      # to avoid //, if only for less confusing output for the user.
      lt_sysroot=`$CC --print-sysroot 2>/dev/null | $SED 's:/\+$::'`
    fi
    ;; #(
+ yes|''|/)
+   ;; #(
  /*)
    lt_sysroot=`echo "$with_libtool_sysroot" | $SED -e "$sed_quote_subst"`
    ;; #(
- no|'')
-   ;; #(
  *)
    AC_MSG_RESULT([$with_libtool_sysroot])
    AC_MSG_ERROR([The sysroot must be an absolute path.])
